src/bootstrap/lib.rs

```diff
@@ -819,6 +829,12 @@ impl Build {
             && !target.contains("msvc")
         {
             Some(self.cc(target))
+        } else if can_use_lld
+            && self.config.use_lld
+            && target.contains("pc-windows-msvc")
+            && self.build == target
```

Comment:
Why do we only use the linker on Windows (msvc)? Even if it doesn't work on other platforms today, we shouldn't *ignore* the option just because of that (at the very least it should be an error).

Is it also true that this could skip the build=target detection since lld is a cross linker naturally, or am I not recalling correctly?

Reply:
My intention is to have `use_lld` on by default, hence the whitelist.

> Is it also true that this could skip the build=target detection since lld is a cross linker naturally, or am I not recalling correctly?

It seems to know how to find the Windows SDK, but I don't know if that works when cross compiling.

Reply:
(Just to make sure -- this PR is not turning it on by default? I don't think it is, nor would I want it to; that seems like something to leave to a separate PR.)

I think I would like to see the whitelist here removed, and instead when/if we do enable it by default, we would do so conditionally (on the same conditions as you have here). That way, if someone knows it works, they can just enable it themselves vs. having to edit bootstrap to get that to happen.

Reply:
> this PR is not turning it on by default?

It's not.

I removed the whitelist.